src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationException.cs

```diff
@@ -33,7 +33,7 @@ public class ValidationException : Exception
         /// <param name="errorMessage">The localized error message</param>
         /// <param name="validatingAttribute">The attribute that triggered this exception</param>
         /// <param name="value">The value that caused the validating attribute to trigger the exception</param>
-        public ValidationException(string errorMessage, ValidationAttribute validatingAttribute, object value)
+        public ValidationException(string? errorMessage, ValidationAttribute? validatingAttribute, object? value)
```

Comment:
I was tempted to suggest trying to make `ValidationAttribute` not nullable here since I don't think we'd have any uses ourselves where it would be null, but with it being a public constructor and with no references to it other than the property setter, this is the right call.

Reply:
Agreed... I'm generally trying to be as conservative as possible - if there's a public code path which results in something being nullable (and in the type being usable), I tend to mark it nullable, even if it doesn't make a lot of sense.

Reply:
One thing to consider: for arguments we can always relax the constraint and allow null in vNext (and user have an option to pass it anyway with `!`), if we mark it as nullable then constraining it will be a breaking change for vNext (for return values relaxation would be making it non-nullable)

Reply:
@krwq agreed about the general logic, but do you feel that going non-nullable is the right choice for this specific case? For these cases where we're annotating legacy APIs with a very long history, there's non-negligible chance of users using these APIs with null, and if the implementation supports that correctly, my tendency is to annotate the actual, supported state of affairs (at least that's the guidance I've received).

Another point is that an overly-relaxed annotation of a parameter as nullable only has the consequence of not flagging a warning in user code which passes null; again, as long as the API handles that correctly that may not be a big concern.